src/jit/Backend.cpp

```diff
@@ -850,6 +850,14 @@ void JITCompiler::compile()
         default: {
             switch (item->asInstruction()->opcode()) {
             case ByteCode::SelectOpcode: {
+                if (reinterpret_cast<Select*>(item->asInstruction()->byteCode())->valueSize() == 16) {
+#ifdef HAS_SIMD
+                    emitSelect128(m_compiler, item->asInstruction(), -1);
+                    break;
+#else /* HAS_SIMD */
+                    RELEASE_ASSERT_NOT_REACHED();
+#endif /* HAS_SIMD */
```

Comment:
This is not where this code should be called. The `emitSelect` can be called from various locations.

This shows how the float variant is called in 32 bit:
https://github.com/Samsung/walrus/blob/main/src/jit/IntMath32Inl.h#L863

Add a new condition here for simd. Also add it to the 64 bit variant.

Reply:
So, because of the current module structure (ie.: files getting included from `Backend.cpp` with no include guards), I would either have to include `SimdInl.h` from `IntMath32Inl.h`, or move the include of `SimdInl.h` above the `IntMath` includes in `Backend.cpp`.  I tried the latter, it resulted in compile errors due to various macros getting redefined.  I can try the former, but frankly I think we should start using include guards, because the current structure makes things needlessly complicated.

Reply:
I mean... I guess that works. :thinking:
I still think include guards would help.  It sucks that I can't use IDE features like "jump to definition".  But whatever, that's for another PR.

Reply:
Maybe the patch was not updated but this check should be in `emitSelect` which is called from various locations.

Reply:
I'm pretty sure I pushed it yesterday.  Github even says that the diff is outdated.